scheduling/tasks.go

```diff
@@ -53,11 +55,16 @@ type TaskMetadata struct {
 	DateCreated time.Time  `json:"dateCreated"`
 }
 
-func (t *TaskMetadata) Marshal() ([]byte, error) {
-	return json.Marshal(t)
+func (t *TaskMetadata) Marshal() ([]byte, fferr.GRPCError) {
+	bytes, err := json.Marshal(t)
+	if err != nil {
+		errMessage := fmt.Errorf("failed to serialize task metadata: %w", err)
+		return nil, fferr.NewInternalError(errMessage)
+	}
+	return bytes, nil
 }
 
-func (t *TaskMetadata) Unmarshal(data []byte) error {
+func (t *TaskMetadata) Unmarshal(data []byte) fferr.GRPCError {
```

Comment:
we don't need to specifically return a GRPCError everywhere, we can just return a regular error

Reply:
since we wrap it anyways we can rely on that